index.js

```diff
@@ -146,18 +149,21 @@ var TruffleContractSchema = {
   // - Resolves as validated `contractObj`
   // - Rejects with list of errors from schema validator
   validate: function(contractObj) {
-    var ajv = new Ajv();
-    var validate = ajv.compile(contractSchema);
-    if (validate(contractObj)) {
+    var ajv = new Ajv({ useDefaults: true });
+    ajv.addSchema(abiSchema);
```

Comment:
Ah, the magic of json-schema! Specifically it's [this `$ref`](https://github.com/trufflesuite/truffle-contract-schema/pull/7/files#diff-76dc6ee27dce76d47ec972f216f2b8cbR8) that does it.